packages/flutter/lib/src/material/material_button.dart

```diff
@@ -291,7 +279,7 @@ class MaterialButton extends StatelessWidget {
 //
 // This class only exists to give the label and icon button widgets a distinct
 /// class for the sake of [ButtonTheme]. It can not be instantiated.
-abstract class MaterialButtonWithIconMixin {
+mixin MaterialButtonWithIconMixin {
   // This class is intended to be used as a mixin, and should not be
   // extended directly.
   factory MaterialButtonWithIconMixin._() => null;
```

Comment:
You can remove the factory constructor trick, since it isn't possible to instantiate a mixin